tiledb/sm/crypto/test/unit_crypto.cc

```diff
@@ -493,50 +522,96 @@ TEST_CASE("Crypto: Test AES-256-GCM", "[crypto][aes]") {
   }
 }
 
+static std::string from_hex(const std::string& str) {
+  std::string output;
+  for (size_t i = 0; i < str.length(); i += 2) {
+    char byte_str[3] = {str[i], str[i + 1], '\0'};
+    output.push_back((uint8_t)std::strtoul(byte_str, nullptr, 16));
+  }
+  return output;
+}
+
+static std::string to_hex(const uint8_t* data, uint64_t length) {
+  std::stringstream ss;
+  for (uint64_t i = 0; i < length; i++) {
+    ss << std::hex << std::setw(2) << std::setfill('0') << (int)data[i];
+  }
+  return ss.str();
+}
+
+/**
+ * Test that the given input (optionally in hex) has the expected hash value in
+ * hex. The function is generic over the hash and the input size.
+ */
+template <Status Hash(const void*, uint64_t, Buffer*), int Digest_Bytes>
+static void test_expected_hash_value(
+    const std::string& input, const std::string& expected_value, bool hex) {
+  REQUIRE(expected_value.length() == Digest_Bytes * 2);
+
+  const std::string& processed_input = hex ? from_hex(input) : input;
+
+  Buffer hash_buf(Digest_Bytes);
+  CHECK(
+      (Hash(processed_input.data(), processed_input.length(), &hash_buf)).ok());
+
+  // Compare the strings for a better error message in case of failure.
+  CHECK(
+      to_hex((uint8_t*)hash_buf.data(), hash_buf.alloced_size()) ==
+      expected_value);
+}
+
 TEST_CASE("Crypto: Test MD5", "[crypto][md5]") {
-  SECTION("- Basic") {
-    std::string expected_checksum = "e99a18c428cb38d5f260853678922e03";
-    std::string text_to_checksum = "abc123";
-    ConstBuffer input_buffer(
-        text_to_checksum.data(), text_to_checksum.length());
-    Buffer output_buffer;
-    CHECK(output_buffer.realloc(Crypto::MD5_DIGEST_BYTES).ok());
-    CHECK(Crypto::md5(&input_buffer, &output_buffer).ok());
-
-    unsigned char* digest =
-        reinterpret_cast<unsigned char*>(output_buffer.data());
-    char md5string[33];
-    for (uint64_t i = 0; i < output_buffer.alloced_size(); ++i) {
-      sprintf(&md5string[i * 2], "%02x", (unsigned int)digest[i]);
-    }
-    CHECK(
-        memcmp(
-            expected_checksum.data(), md5string, expected_checksum.length()) ==
-        0);
+  auto test_md5 =
+      test_expected_hash_value<Crypto::md5, Crypto::MD5_DIGEST_BYTES>;
+  static const std::vector<std::pair<std::string, std::string>> test_cases{
```

Comment:
`string_view` can be initialized with string literals and converts readily to `span`.